python/semantic_kernel/connectors/ai/anthropic/services/anthropic_chat_completion.py

```diff
@@ -0,0 +1,238 @@
+from typing import Any
+
+from anthropic import AsyncAnthropic
+from anthropic.types import (
+    ContentBlockStopEvent,
+    Message,
+    RawContentBlockDeltaEvent,
+    RawMessageDeltaEvent,
+    RawMessageStartEvent,
+    TextBlock,
+)
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.ai.anthropic.prompt_execution_settings.anthropic_prompt_execution_settings import (
+    AnthropicChatPromptExecutionSettings,
+)
+from semantic_kernel.connectors.ai.anthropic.settings.anthropic_settings import AnthropicSettings
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.connectors.ai.prompt_execution_settings import PromptExecutionSettings
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.chat_message_content import ITEM_TYPES, ChatMessageContent
+from semantic_kernel.contents.streaming_chat_message_content import ITEM_TYPES as STREAMING_ITEM_TYPES
+from semantic_kernel.contents.streaming_chat_message_content import StreamingChatMessageContent
+from semantic_kernel.contents.streaming_text_content import StreamingTextContent
+from semantic_kernel.contents.text_content import TextContent
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.exceptions.service_exceptions import (
+    ServiceInitializationError,
+    ServiceResponseException,
+)
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+from .utils import ANTHROPIC_TO_SEMANTIC_KERNEL_FINISH_REASON_MAP, AnthropicFinishReason
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+@experimental_class
+class AnthropicChatCompletion(ChatCompletionClientBase):
+    """Antropic ChatCompletion class."""
+
+    async_client: AsyncAnthropic
+    
+    def __init__(
+        self,
+        ai_model_id: str | None = None,
+        service_id: str | None = None,
+        api_key: str | None = None,
+        async_client: AsyncAnthropic | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+    ) -> None:
+        """Initialize an AnthropicChatCompletion service.
+
+        Args:
+            ai_model_id (str): Anthropic model name, see
+                https://docs.anthropic.com/en/docs/about-claude/models#model-names
+            service_id (str | None): Service ID tied to the execution settings.
+            api_key (str | None): The optional API key to use. If provided will override,
+                the env vars or .env file value.
+            async_client (AsyncAnthropic | None) : An existing client to use. 
+            env_file_path (str | None): Use the environment settings file as a fallback
+                to environment variables. 
+            env_file_encoding (str | None): The encoding of the environment settings file. 
+        """
+        try:
+            anthropic_settings = AnthropicSettings.create(
+                api_key=api_key,
```

Comment:
I think we'd need something like: 

```
if not async_client and not anthropic_settings.api_key:
    raise ServiceInitializationError("The Anthropic API key is required.")
```

What do you think?

Reply:
Ah! Very good. I missed that the `api_key` in the anthropic settings is required. Thanks.

Reply:
I think that the `api_key` in the settings _should not be_ required. If someone passes in a custom client, they don't need to also pass in the API key. That's when this should be used:

```
if not async_client and not anthropic_settings.api_key:
    raise ServiceInitializationError("The Anthropic API key is required.")
```

Reply:
right now it seems like this current implementation is consistent across chatcompletion service implementations (eg. [google](https://github.com/microsoft/semantic-kernel/blob/bdf15a80ed2e303aea56fa95835400380055f10f/python/semantic_kernel/connectors/ai/google/google_ai/services/google_ai_chat_completion.py#L90C9-L98C99), mistral, etc) and each respective settings object requires the api key.